File: Data Structures & Algorithms/duplicate-integer/submission-3.py
class Solution:
    def hasDuplicate(self, nums: List[int]) -> bool:
        nums = sorted(nums)
        length = len(nums)
        i = 0 
        j = 1

        if length < 2:
            return False
        
        while i < length:
            while j < length:
                if nums[i] != nums[j]:
                    i += 1
                    j += 1
                else:
                    return True
            return False
        

File: Data Structures & Algorithms/two-integer-sum/submission-3.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> indices = new Dictionary<int, int>();

        for (int i = 0; i < nums.Length; i++){
            int twoSum = target - nums[i];
            if (indices.ContainsKey(twoSum)){
                return new int[] { indices[twoSum], i};
            }
            indices[nums[i]] = i;
        }

        throw new InvalidOperationException("No valid pair found.");
    }
}
